src/Microsoft.TestPlatform.CrossPlatEngine/TestSession/ProxyTestSessionManager.cs

```diff
@@ -382,6 +384,35 @@ private void DisposeProxies()
             _proxyMap.Clear();
         }
     }
+
+    private bool CheckRunSettingsAreCompatible(string requestRunSettings)
+    {
+        if (_runSettingsEnvironmentVariables.Count == 0)
+        {
+            _runSettingsEnvironmentVariables = InferRunSettingsHelper.GetEnvironmentVariables(
```

Comment:
does that mean that if the session is created with valid runsettings, but a discovery request with invalid runsettings, we'll default to session's runsettings? it might be more clear if an error would be thrown then

Reply:
Depends on what you mean by invalid runsettings. Invalid in that they're not compatible when matched against the original test session runsettings ? If that's the case we don't default to the original test session runsettings but this method will return false and one layer above, in `DequeProxy` we throw an `InvalidOperationException`.